apps/ledger-live-mobile/src/screens/Manager/index.tsx

```diff
@@ -83,132 +47,100 @@ type Props = {
 
 type ChooseDeviceProps = Props & {
   isFocused: boolean;
-  removeKnownDevice: (_: string) => void;
 };
 
-class ChooseDevice extends Component<
-  ChooseDeviceProps,
-  {
-    showMenu: boolean;
-    device?: Device;
-    result?: any;
-  }
-> {
-  state = {
-    showMenu: false,
-    device: undefined,
-    result: undefined,
-  };
+const ChooseDevice: React.FC<ChooseDeviceProps> = ({ isFocused }) => {
+  const [device, setDevice] = useState<Device | undefined>();
 
-  chosenDevice: Device;
+  const [chosenDevice, setChosenDevice] = useState<Device | undefined>();
+  const [showMenu, setShowMenu] = useState<boolean>(false);
 
-  onShowMenu = (device: Device) => {
-    this.chosenDevice = device;
-    this.setState({ showMenu: true });
-  };
+  const navigation = useNavigation<BaseNavigatorProps>();
+  const { params } = useRoute<{
+    params: RouteParams;
+    name: string;
+    key: string;
+  }>();
 
-  onHideMenu = () => {
-    this.setState({ showMenu: false });
-  };
+  const newDeviceSelectionFeatureFlag = useFeature("llmNewDeviceSelection");
 
-  onSelectDevice = (device?: Device) => {
+  const onSelectDevice = (device?: Device) => {
     if (device)
       track("ManagerDeviceEntered", {
         modelId: device.modelId,
       });
-    this.setState({ device });
+    setDevice(device);
+  };
+
+  const onShowMenu = (device: Device) => {
+    setChosenDevice(device);
+    setShowMenu(true);
   };
 
-  onSelect = (result: any) => {
-    this.setState({ device: undefined, result });
-    const {
-      route: { params = {} },
-    } = this.props;
+  const onHideMenu = () => setShowMenu(false);
+
+  const onSelect = (result: any) => {
```

Comment:
Please do not use `any`.

Reply:
Using `unknown` instead as the result type from a device action is very "unknown"